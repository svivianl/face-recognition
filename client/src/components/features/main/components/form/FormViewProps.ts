interface FormViewProps {
  buttonText: string;
  buttonDisabled: boolean;
  url: string;
  error?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onSubmit: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

export default FormViewProps;
