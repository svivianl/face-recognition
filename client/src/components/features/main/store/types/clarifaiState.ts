export type ErrorType = "face-recognition";

export interface MessageError {
  type: ErrorType;
  message: string;
}

export type StatusType = "saved" | "updated" | "deleted" | "error";

export interface Status {
  type: StatusType;
  error?: MessageError;
}

export interface ClarifaiState {
  isLoading: boolean;
  url: string;
  regions: any;
  status: Status | null;
}
