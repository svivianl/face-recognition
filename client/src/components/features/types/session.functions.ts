export const saveAuthToken = (token: string) => {
  window.sessionStorage.setItem("token", token);
};
