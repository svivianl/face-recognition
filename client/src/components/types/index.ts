export * from "./particles";
export * from "./user";
export * from "./user.functions";
