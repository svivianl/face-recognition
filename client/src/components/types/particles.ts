export const particlesParams = {
  particles: {
    number: {
      value: 200,
      density: {
        enable: true,
        value_area: 1000,
      },
    },
  },
};
