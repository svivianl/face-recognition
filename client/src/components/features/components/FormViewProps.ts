import { ReactNode } from "react";
import { User, SignIn } from "../../../types";

interface FormViewProps {
  name: string;
  buttonText: string;
  buttonDisabled: boolean;
  user: User | SignIn;
  inputError: User | SignIn;
  error?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onSubmit: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  children?: ReactNode;
}

export default FormViewProps;
