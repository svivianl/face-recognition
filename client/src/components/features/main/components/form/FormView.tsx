import React from "react";
import FormViewProps from "./FormViewProps";
import "../../../../../css/features/main/components/form/form.css";

const FormView = ({
  buttonText,
  buttonDisabled,
  url,
  error,
  onChange,
  onSubmit,
}: FormViewProps) => {
  return (
    <div className="border border-dark rounded url-form-container form-container mt-3">
      <form>
        <div className="form-group">
          <div className="form-input">
            <label htmlFor="url" className="mr-2">
              URL<span className="text-danger">*</span>
            </label>
            <input
              type="url"
              className="form-control d-inline-block"
              id="url"
              placeholder="https://samples.clarifai.com/face-det.jpg"
              onChange={onChange}
              value={url}
            />
            <button
              type="submit"
              className="btn btn-primary"
              onClick={onSubmit}
              disabled={buttonDisabled}
            >
              {buttonText}
            </button>
          </div>
          {error && <div className="text-danger">{error}</div>}
        </div>
      </form>
    </div>
  );
};

export default FormView;
