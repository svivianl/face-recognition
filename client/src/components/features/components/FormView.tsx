import React from "react";
import FormViewProps from "./FormViewProps";
import "../../../css/features/form.css";

const FormView = ({
  name,
  buttonText,
  buttonDisabled,
  user,
  inputError,
  error,
  onChange,
  onSubmit,
  children,
}: FormViewProps) => {
  return (
    <div className="border border-dark rounded form-container mt-5">
      <div className="mb-5">
        <h1 className="text-center">{name}</h1>
        {error && (
          <p id="form-error" className="text-danger text-center mb-3 mt-3">
            {error}
          </p>
        )}
      </div>
      <form>
        {children}
        <div className="form-group">
          <div className="form-input">
            <label htmlFor="email" className="mr-2">
              E-mail<span className="text-danger">*</span>
            </label>
            <input
              type="email"
              className="form-control d-inline-block"
              id="email"
              placeholder="name@example.com"
              onChange={onChange}
              value={user.email}
            />
          </div>
          {inputError && inputError.email && (
            <div>
              <p id="error-email" className="text-danger">
                {inputError.email}
              </p>
            </div>
          )}
        </div>
        <div className="form-group column">
          <div className="form-input">
            <label htmlFor="password" className="mr-2">
              Password<span className="text-danger">*</span>
            </label>
            <input
              className="form-control"
              id="password"
              type="password"
              onChange={onChange}
              value={user.password}
            />
          </div>
          {inputError && inputError.password && (
            <div>
              <p id="error-password" className="text-danger">
                {inputError.password}
              </p>
            </div>
          )}
        </div>
        <div className="col text-center">
          <button
            type="submit"
            className="btn btn-primary"
            onClick={onSubmit}
            disabled={buttonDisabled}
          >
            {buttonText}
          </button>
        </div>
      </form>
    </div>
  );
};

export default FormView;
