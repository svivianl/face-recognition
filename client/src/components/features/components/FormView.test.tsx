import React from "react";
import { mount, shallow } from "enzyme";
import FormView from "./FormView";
import FormViewProps from "./FormViewProps";
import { User } from "../../../types";

describe("FormView component", () => {
  let onChangeMock: any = jest.fn();
  let onSubmitMock: any = jest.fn();
  let mockProps: FormViewProps;

  beforeEach(() => {
    onChangeMock.mockClear();
    onSubmitMock.mockClear();

    mockProps = {
      name: "Sign In",
      buttonText: "Sign in",
      buttonDisabled: false,
      user: {} as User,
      onChange: onChangeMock,
      onSubmit: onSubmitMock,
    };
  });

  it("should render", () => {
    expect.assertions(5);
    const wrapper = shallow(<FormView {...mockProps} />);

    expect(wrapper).toMatchSnapshot();

    const element = wrapper.find(`button`);
    expect(element.length).toBe(1);
    expect(element.props().disabled).toBeFalsy();

    const elementError = wrapper.find(`h1`);
    expect(elementError.length).toBe(1);
    expect(elementError.text()).toBe(mockProps.name);
  });

  describe("input", () => {
    describe("render error", () => {
      it("should display 'email' is mandatory", () => {
        expect.assertions(4);

        const error: User = { email: "Please insert your email address" };
        const wrapper = mount(
          <FormView {...mockProps} inputError={error} buttonDisabled={true} />
        );

        const element = wrapper.find(`button`);
        expect(element.length).toBe(1);
        expect(element.props().disabled).toBeTruthy();

        const elementError = wrapper.find(`[id="error-email"]`);
        expect(elementError.length).toBe(1);
        expect(elementError.text()).toBe(error.email);
      });

      it("should display 'password' is mandatory", () => {
        expect.assertions(4);

        const error: User = { password: "Please insert a password" };
        const wrapper = mount(
          <FormView {...mockProps} inputError={error} buttonDisabled={true} />
        );

        const element = wrapper.find(`button`);
        expect(element.length).toBe(1);
        expect(element.props().disabled).toBeTruthy();

        const elementError = wrapper.find(`[id="error-password"]`);
        expect(elementError.length).toBe(1);
        expect(elementError.text()).toBe(error.password);
      });
    });

    it("should change email", () => {
      expect.assertions(2);

      const input = "john@doe.com";
      const mockEvent = {
        preventDefault() {},
        target: { value: input },
      } as React.ChangeEvent<HTMLInputElement>;
      const wrapper = mount(<FormView {...mockProps} />);
      const component = wrapper.find(`[id="email"]`);

      expect(component.length).toBe(1);
      component.at(0).simulate("change", mockEvent);
      expect(onChangeMock.mock.calls.length).toBe(1);
    });

    it("should change password", () => {
      expect.assertions(2);

      const input = "1234567";
      const mockEvent = {
        preventDefault() {},
        target: { value: input },
      } as React.ChangeEvent<HTMLInputElement>;
      const wrapper = mount(<FormView {...mockProps} />);
      const component = wrapper.find(`[id="password"]`);

      expect(component.length).toBe(1);
      component.at(0).simulate("change", mockEvent);
      expect(onChangeMock.mock.calls.length).toBe(1);
    });
  });

  it("should disable submit button when 'error' is not empty", () => {
    expect.assertions(4);
    const error = "Please insert a valid URL";
    const wrapper = mount(
      <FormView {...mockProps} error={error} buttonDisabled={true} />
    );

    const element = wrapper.find(`button`);
    expect(element.length).toBe(1);
    expect(element.props().disabled).toBeTruthy();

    const elementError = wrapper.find(`[id="form-error"]`);
    expect(elementError.length).toBe(1);
    expect(elementError.text()).toBe(error);
  });

  it("should render with error", () => {
    expect.assertions(2);

    const error = "Request failed with status code 400";
    const wrapper = mount(<FormView {...mockProps} error={error} />);
    const element = wrapper.find(`[id="form-error"]`);

    expect(element.length).toBe(1);
    expect(element.text()).toBe(error);
  });

  it("should submit form", () => {
    expect.assertions(2);

    const user = {
      email: "john@doe.com",
      password: "1234567",
    };
    const wrapper = mount(<FormView {...mockProps} user={user} />);
    const element = wrapper.find(`button`);
    expect(element.length).toBe(1);
    element.at(0).simulate("click");
    expect(onSubmitMock.mock.calls.length).toBe(1);
  });

  it("should change submit button when submitting", () => {
    expect.assertions(2);

    const wrapper = mount(
      <FormView {...mockProps} buttonDisabled={true} buttonText="Submitting" />
    );
    const element = wrapper.find(`button`);

    expect(element.length).toBe(1);
    expect(element.props().disabled).toBeTruthy();
  });
});
