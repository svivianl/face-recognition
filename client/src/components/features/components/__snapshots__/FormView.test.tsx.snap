// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormView component should render 1`] = `ShallowWrapper {}`;
